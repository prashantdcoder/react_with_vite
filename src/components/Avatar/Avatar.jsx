/* eslint-disable no-unused-vars */
import React from "react";
import PropTypes from "prop-types";

const Avatar = ({ url }) => {
  return (
    <div className="avatar-container">
      <img loading="lazy" src={url} alt="avatar_img" />;
    </div>
  );
};

Avatar.propTypes = {
  url: PropTypes.string.isRequired,
};

export default Avatar;
