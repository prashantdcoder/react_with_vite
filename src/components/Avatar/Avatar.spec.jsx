/* eslint-disable no-undef */
import "@testing-library/jest-dom";
import React from "react";
import { render } from "@testing-library/react";
import "react";
import Avatar from "./Avatar";

describe("Avatar Component", () => {
  test("renders correctly", () => {
    const { asFragment } = render(<Avatar url={"test"} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
