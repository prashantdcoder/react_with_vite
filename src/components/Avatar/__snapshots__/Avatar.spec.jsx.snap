// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="avatar-container"
  >
    <img
      alt="avatar_img"
      loading="lazy"
      src="test"
    />
    ;
  </div>
</DocumentFragment>
`;
